### ISSUE TITLE
loss nan in VAE training

### ISSUE THREAD

**Issue by shwj114514** [Stability-AI/stable-audio-tools#144]
Thank you for your excellent work and the well-designed open-source code.

When I use your training code to train from scratch, I frequently encounter a situation where the loss becomes NaN after a certain number of training steps. Is this behavior expected?
<img width="2436" alt="image" src="https://github.com/user-attachments/assets/9dae6db2-3898-45d1-ad75-7e91ca78bd12">

This issue occurs when training both 44100 mono and stereo audio files. I have to repeat the training multiple times to ensure the loss remains stable.
<img width="2195" alt="image" src="https://github.com/user-attachments/assets/2f84f2ed-e597-49b1-92c1-2bc3b1bcadc0">

I am using the  [stable audio open config](https://github.com/Stability-AI/stable-audio-tools/blob/main/stable_audio_tools/configs/model_configs/autoencoders/stable_audio_2_0_vae.json).

**Comment by Kami-prog** [comment]
maybe this will help
https://bit.ly/3XQLHlH

you may need to install the c compiler

**Comment by apply74** [comment]
I also encountered this problem. When I increased the model parameters, the training was unstable. Is it going to be solved?

**Comment by shwj114514** [comment]
> I also encountered this problem. When I increased the model parameters, the training was unstable. Is it going to be solved?

I solved this problem by reducing the learning rates of both the generator and discriminator to 1/10 of their original values, and the training became stable.

**Comment by apply74** [comment]
I also tried reducing the learning rate. Although the training is stable, the reconstruction result will be very poor.

**Comment by fletcherist** [comment]
<img width="2011" alt="Screenshot 2024-09-26 at 11 41 49" src="https://github.com/user-attachments/assets/224d9b59-d070-4479-aaea-5e2986460ee7">

the same thing

**Comment by apply74** [comment]
> <img alt="Screenshot 2024-09-26 at 11 41 49" width="2000" src="https://private-user-images.githubusercontent.com/15146860/371056453-224d9b59-d070-4479-aaea-5e2986460ee7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczNDA0OTgsIm5iZiI6MTcyNzM0MDE5OCwicGF0aCI6Ii8xNTE0Njg2MC8zNzEwNTY0NTMtMjI0ZDliNTktZDA3MC00NDc5LWFhZWEtNWUyOTg2NDYwZWU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI2VDA4NDMxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExNDNhNTQ1MTgzMjg1Y2EzZGYxN2E5N2IwYTZmMzM5ZDA1ZjcxZWQ2ZjA3NmE3ZjFlZTI5NjJkZmM5M2E5NWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xwdViD6C2U7pejrjavDoDfFPKYfy510BQ03AoaAdHq8">
> the same thing
I have solved the problem by increating the batch_size from 1 to 5.

**Comment by fletcherist** [comment]
> > <img alt="Screenshot 2024-09-26 at 11 41 49" width="2000" src="https://private-user-images.githubusercontent.com/15146860/371056453-224d9b59-d070-4479-aaea-5e2986460ee7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczNDA0OTgsIm5iZiI6MTcyNzM0MDE5OCwicGF0aCI6Ii8xNTE0Njg2MC8zNzEwNTY0NTMtMjI0ZDliNTktZDA3MC00NDc5LWFhZWEtNWUyOTg2NDYwZWU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI2VDA4NDMxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExNDNhNTQ1MTgzMjg1Y2EzZGYxN2E5N2IwYTZmMzM5ZDA1ZjcxZWQ2ZjA3NmE3ZjFlZTI5NjJkZmM5M2E5NWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xwdViD6C2U7pejrjavDoDfFPKYfy510BQ03AoaAdHq8">
> > the same thing
> 
> I have solved the problem by increating the batch_size from 1 to 5.

oh rly? let me try it but i think this batch size doesn't fit to gpu))
i'll message here after a try. thanks for your help very appreciate it

**Comment by fletcherist** [comment]
> reducing the learning rates of both the generator and discriminator to 1/10 of their original values

this works

**Comment by nateraw** [comment]
You have to tune the learning rates. Higher batch size helps keep things stable.

Another tip is if you can't get large enough batch size, you can reduce the sample size which should free up enough memory to bump back up the batch size.

Hope this helps ❤️

**Comment by stg1205** [comment]
Also I noticed that doing vad to remove silence part help
